### PR TITLE
feat: aggregate array cursors

### PR DESCRIPTION
**Pull request by jlapacik**

### CODE REVIEW

File: storage/reads/array_cursor.go [hunk 1]
```diff
@@ -38,131 +51,16 @@ func newWindowAggregateArrayCursor(ctx context.Context, agg *datatypes.Aggregate
...
-			return &m.cursors.b
-		default:
-			panic(fmt.Sprintf("unreachable: %T", cur))
-		}
-	}
-	// if cursor is nil
-	return nil
-}
-
-func newSumArrayCursor(cur cursors.Cursor) cursors.Cursor {
```
Comment (jlapacik) [inline storage/reads/array_cursor.go:122]: This is subsumed by WindowAggregateArrayCursor now.

File: storage/reads/array_cursor.go [hunk 2]
```diff
@@ -38,131 +51,16 @@ func newWindowAggregateArrayCursor(ctx context.Context, agg *datatypes.Aggregate
...
-		return newIntegerArraySumCursor(cur)
-	case cursors.UnsignedArrayCursor:
-		return newUnsignedArraySumCursor(cur)
-	default:
-		// TODO(sgc): propagate an error instead?
-		return nil
-	}
-}
-
-func newCountArrayCursor(cur cursors.Cursor) cursors.Cursor {
```
Comment (jlapacik) [inline storage/reads/array_cursor.go:136]: This is also subsumed by WindowAggregateArrayCursor now.

File: storage/reads/array_cursor.go [hunk 3]
```diff
@@ -38,131 +51,16 @@ func newWindowAggregateArrayCursor(ctx context.Context, agg *datatypes.Aggregate
 	}
 }
 
-type arrayCursors struct {
```
Comment (jlapacik) [inline storage/reads/array_cursor.go:41]: This isn't used at all. OSS uses `multiShardArrayCursors` which is almost identical in structure except for the underlying series cursor.

File: storage/reads/array_cursor.go [hunk 4]
```diff
@@ -38,131 +51,16 @@ func newWindowAggregateArrayCursor(ctx context.Context, agg *datatypes.Aggregate
...
+	itrs cursors.CursorIterators
+	err  error
 }
 
 type multiShardArrayCursors struct {
-	ctx   context.Context
-	limit int64
-	req   cursors.CursorRequest
+	ctx context.Context
+	req cursors.CursorRequest
```
Comment (jlapacik) [inline storage/reads/array_cursor.go:63]: Aggregation is now handled by WindowAggregateArrayCursor, so there's no need to calculate summary stats like count and limit here.

**Review by jsternberg — APPROVED** [review]

**Pull request merged**